source/horoscope/js/horoscope.js

```diff
@@ -48,6 +48,7 @@ async function init() {
         let category = categoryElement.value;
         let message = fortuneElement.innerText;
         let today = new Date();
+        today.setHours(0,0,0,0);
```

Comment:
Why do we need this?  This sets the hours in the date object to 00:00:00

Causes the "min ago" of the past history to be wrong

Reply:
i removed this line. i am excluding the date field when checking if the horoscope exists in the saveHoroscope function itself now. The time function looks like it is working correctly now.